src/transformers/models/swin/modeling_swin.py

```diff
@@ -1214,3 +1224,99 @@ def forward(
             attentions=outputs.attentions,
             reshaped_hidden_states=outputs.reshaped_hidden_states,
         )
+
+
+@add_start_docstrings(
+    """
+    Swin backbone, to be used with frameworks like DETR and MaskFormer.
+    """,
+    SWIN_START_DOCSTRING,
+)
+class SwinBackbone(SwinPreTrainedModel):
+    def __init__(self, config: SwinConfig):
+        super().__init__(config)
+
+        self.stage_names = config.stage_names
+        # make sure model outputs hidden states before downsampling
+        config.output_hidden_states_before_downsampling = True
```

Comment:
No, we don't force-change a config value when instantiating a model. If you do this, there is no real choice for the user. It looks like you need an argument to the forward method of the encoder instead, which would decide whether or not to downsample the hidden states.
Also can we call it `downsample_hidden_states` to be shorter?

Reply:
But when you add an argument to the forward of the encoder, you also have to add it to the forward of `SwinModel` right?

Reply:
I don't see why?

Reply:
The backbone class doesn't have to use Swin model though, it can reimplement the forward to use the encoder.

Reply:
Hmm that would be inconsistent with other backbone classes. Ideally, each Backbone class uses the base model, e.g. `ConvNextBackbone` uses `self.convnext` = `ConvNextModel` as an attribute. This way, people can leverage the from_pretrained method on backbones, which will ensure that all weights are loaded correctly.

If backbones were to use the encoder classes, this wouldn't work due to the `base_model_prefix` not matching

Reply:
So to me it seems the only solution is an attribute of the config..

Reply:
I don't see why the backbone class *has to* use the base model, especially when it does different things than the base model. It doesn't make any sense to add a config flag for different behavior (which goes against the Transformers philosophy) just so you can use the base model in the backbone.

And `from_pretrained` will work fine whether or not you have a base model attribute.

Reply:
If you want at some point, if it's used in a larger checkpoint (in a model using resnet as a backbone) it might be breaking though.

Reply:
Yes we can still update `ResNetBackbone` as it's not being used yet